pkg/dao/utils_test.go

```diff
@@ -0,0 +1,68 @@
+package dao
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/content-services/content-sources-backend/pkg/config"
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"github.com/content-services/content-sources-backend/pkg/seeds"
+	uuid2 "github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"gorm.io/gorm"
+)
+
+func createRepository(t *testing.T, tx *gorm.DB, prefix string, redhatRepo bool) models.RepositoryConfiguration {
+	const lookup string = "0123456789abcdefghijklmnopqrstuvwxyz"
+	randomName := seeds.RandStringWithChars(10, lookup)
+	URL := "https://example.com/" + randomName
+	origin := config.OriginExternal
+	name := "toSnapshot" + prefix + randomName
+	orgID := "someOrg"
+	if redhatRepo {
+		URL = "https://example.redhat.com"
+		origin = ""
```

Comment:
thanks for DRYing up these helpers! this looks great and should get rid of those invalid uuid errors in glitchtip :smile: 

one thing i see: this was actually missing in the original helper and not related to this ticket so feel free to ignore, but if you want to fix it here, the origin field here for red hat repos should be set to `config.OriginRedHat` to mimic actual behavior. this change would require some other fixes in TestCreateAndListRedHatRepo to filter correctly too